### DISCUSSION TITLE
Frigate dual NIC setup: (block cameras from accessing the internet)[Question]:

### DISCUSSION THREAD

**Discussion by lukasig** [blakeblackshear/frigate#10469] [Ask A Question] [answered]
### What is your question:

This is not frigate specific, but it is related for those who want to stop their cameras from accessing the internet for enhanced privacy or other reasons.

Following the official frigate docs on [recommended hardware](https://docs.frigate.video/frigate/hardware/#server), I would appreciate some guidelines of how to setup a dedicated private network for my cameras where they can be blocked from accessing the internet when using a dual nic server.

I have a Beelink EQ12 Pro with dual nics running Debian.
On another figate setup I use a Acemagic S1 with dual nics also running Debian.

I would prefer this to be done entirely on the server side (including DHCP) without the need of changing main router configs.

thanks in advance.

**Comment by blakeblackshear** [comment, marked as answer]
Set static IPs on the cameras and the second NIC so they are in the same subnet. Then simply connect them together with a switch. I use 10.10.0.0/24 for the cameras and my main network is 192.168.2.0/24.

**Reply by lukasig** [reply]
Thank you

Makes enough sense even for me to have a starting point.
I will tackle this sometime next week and maybe will come back to this thread with a post of a step by step guide, just to make it easier for those who might need help in this regard.

and a sincere **THANK YOU** for all your hard work @blakeblackshear

**Reply by darnmason** [reply]
Sorry to resurrect an old discussion, but I have 2 NICs and was wondering the same. I assume it's not possible to access the camera's management web interface with this setup? Or is there some router magic that can make it work from the main network.

**Reply by NickM-27** [reply]
you can use nginx or another proxy to make this available

**Reply by darnmason** [reply]
Ah I just setup nginx last night, I can see how that would work now, thanks a lot for the reply!

**Reply by kwsabq** [reply]
Im sorry to jump onto this question after it has been closed, but I'm setting up a beelink eq13 with dual NICs and was hoping you would share your experiace on how you set up the IP addresses and such @lukasig . Im struggling to get my cameras and everything working properly.

**Reply by amm009uk** [reply]
I thought nginx would be for inbound connections but cams can still connect out?
For those of us without dual NICs, I was hoping to isolate cams with VLANs?

**Reply by nukefrenzy** [reply]
I'm no linux expert, so I had to do a bit of searching to figure out how to do this.  Hopefully this will help the next person.  Everything is working great, but maybe there's an issue I should fix or a better way?  Let me know.

First off, I have a BeeLink EQi12 with dual NICs running Debian 12 (headless, no graphical interface) using the [Frigate docs](https://docs.frigate.video/guides/getting_started#install-debian-12-bookworm) to install it.  It was already running great and integrated into Home Assistant, but I wanted to move the cameras to the second NIC.
Then I did the following:
1) Disable ipv6, just to remove even the remote possibility of that messing with what I'm trying to do, using this guide (sysctl method): https://itslinuxfoss.com/disable-ipv6-debian-12/
2) found out what my network intefaces are called using this guide: https://vitux.com/debian-network-interface-list/
Mine were called enp2s0 and enp3s0.  Only enp2s0 had an ip address assigned so I knew to use the other one for a static IP.
3) Set a static IP address for the second NIC using this guide: https://devtutorial.io/how-to-configure-static-ip-address-on-debian-12-p3121.html
Specifically, I added the following to the interfaces file:
auto enp3s0
iface enp3s0 inet static
address 10.10.10.1
netmask 255.255.255.0
Notice I did not add a gateway since there's nowhere to go in this configuration.
4) Then I logged into each camera and set their static IP address to the following:
ip address: 10.10.10.XX  where XX is different for each camera, making sure to never use 10.10.10.1 since that's for the BeeLink
mask: 255.255.255.0
gateway: 10.10.10.1
primary dns: blank, or all zeros, whatever your interface requires
backup dns: blank, or all zeros, whatever your interface requires
You probably don't need to specify a gateway here either, but this worked for me.
5) Power everything off, plug in everything to their final spots, and start everything up
6) Log into frigate and edit the config to the new camera IP addresses.  Restart and back to working the way it was before.

**Reply by blakeblackshear** [reply]
I also run an ntp server in my compose file alongside frigate like this so the camera datestamps don't drift:
```yaml
  ntp:
    image: cturra/ntp:latest
    container_name: ntp
    restart: always
    ports:
      - 10.0.11.1:123:123/udp # <---- this should be the static ip on the NIC with the cameras
    environment:
      - NTP_SERVERS=time.cloudflare.com
      - LOG_LEVEL=0
      - TZ=America/Chicago
```

you then set the NTP server on the cameras to the static IP as well

**Reply by nukefrenzy** [reply]
I'm sure I would have ended up noticing this a few months from now.  Thanks for the help!
